src/tests/test_magic.py

```diff
@@ -187,8 +187,17 @@ def test_persist_no_index(ip):
     assert persisted == [(1, "foo"), (2, "bar")]
 
 
-def test_unrecognized_arguments_cell_magic(ip):
-    result = ip.run_cell("%%sql --stuff \n SELECT * FROM test")
+@pytest.mark.parametrize(
+    "sql_statement",
+    [
+        "%%sql --stuff\n SELECT * FROM test",
+        "%%sql --unknown\n SELECT * FROM test",
+        "%%sql --invalid-arg\n SELECT * FROM test",
+        "%sql select * from penguins.csv limit --some sql comment",
```

Comment:
This test case should not give an error.  The logic I mentioned earlier is slightly flawed. Leave the first token aside, as long as you encounter `--` in the token keep continuing, if you stop `break`

Reply:
which test case you refer to? edublancas points out we only need to validate arguments in the cell magic, and the test should only apply to cell magic?

Reply:
> This test case should not give an error. The logic I mentioned earlier is slightly flawed. Leave the first token aside, as long as you encounter `--` in the token keep continuing, if you stop `break`

which test case you refer to? edublancas points out we only need to validate arguments in the cell magic, and the test should only apply to cell magic.

Reply:
`%sql select * from penguins.csv limit --some sql comment`
This shouldn't throw the error

Reply:
Ok, please correct me if I misunderstand:
- This PR is only for `%%sql`, so the command `%sql select * FROM penguins.csv --some` should work.

For this, you must check whether the line is cell or line magic.

Going back to the [comment](https://github.com/ploomber/jupysql/issues/531#issuecomment-1571130589) on the issue; this PR logic can also be implemented for line magic.  See the [comment](https://github.com/ploomber/jupysql/pull/561#discussion_r1220225898) earlier

Reply:
Yes, [this PR](https://github.com/ploomber/jupysql/pull/561#discussion_r1218469471) is only for %%sql. @edublancas is that correct?
I add the check condition whether it is cell or line magic before I check for the unrecognized arguments.

Reply:
> I add the check condition whether it is cell or line magic before I check for the unrecognized arguments.

Command: `%sql select * FROM penguins.csv --some`
I tested it on the binder and it is not working.

Reply:
> > I add the check condition whether it is cell or line magic before I check for the unrecognized arguments.
> 
> Command: `%sql select * FROM penguins.csv --some` I tested it on the binder and it is not working.

It works now

Reply:
Nice, good work.
Just some minor edit: You can pass multiple arguments to pytest parameterized. So instead of coding into the function, you can pass both `sql_statement` and `result` to `pytest.parametrize`

Reply:
Sure, update it now